src/index.js

```diff
@@ -36,22 +25,88 @@ btnDeleteAll.setAttribute('class', 'delete-all');
     const node = document.createElement('li');
-
     node.setAttribute('class', 'todo-item');
     node.setAttribute('data-key', todoItems[i].index);
 
     node.innerHTML = ` 
         <input class="checkbox" id="${todoItems[i].index}" type="checkbox"/>
-        <label for="${todoItems[i].index}" class="option"><img class="select"  src=${select2} alt=""/> <img class="delete"  src=${delete2} alt=""/></label>
+        <label for="${todoItems[i].index}" class="option">
+            <img class="select" src=${select2} alt=""/>
+            <img class="delete" src=${delete2} alt=""/>
+            <img class="edit" src=${edit} alt="edittext"/>
+        </label>
         <span class="items">${todoItems[i].text}</span>
-    
-        `;
+    `;
     ul.append(node);
+
+    // Get the selectdots, deleteoption, and editoption elements for this task
+    const selectdots = node.querySelector('.select');
+    const deleteoption = node.querySelector('.delete');
+    const editoption = node.querySelector('.edit');
+
+    // Add event listeners to show the delete/edit options when the user clicks on the selectdots
+    selectdots.addEventListener('click', () => {
+      selectdots.style.display = 'none';
+      deleteoption.style.display = 'block';
+      editoption.style.display = 'block';
+    });
+
+    // Add event listener for deleting a task
+    deleteoption.addEventListener('click', () => {
+      const itemKey = parseInt(node.getAttribute('data-key'), 10);
+      const itemIndex = todoItems.findIndex((item) => item.index === itemKey);
+      todoItems.splice(itemIndex, 1);
+
+      // Re-index the remaining tasks
+      for (let i = itemIndex; i < todoItems.length; i += 1) {
+        todoItems[i].index = i + 1;
+      }
+
+      localStorage.setItem('todoItems', JSON.stringify(todoItems));
+      display();
+    });
+
+    // Add event listener for editing a task
+    editoption.addEventListener('click', () => {
+      const itemKey = parseInt(node.getAttribute('data-key'), 10);
+      const itemIndex = todoItems.findIndex((item) => item.index === itemKey);
+      const currentTaskText = todoItems[itemIndex].text; // get the current task text
+      const newTaskText = prompt('Please enter the new task description:', currentTaskText); // pre-fill prompt with current task text
+      if (newTaskText !== null && newTaskText !== '') {
+        todoItems[itemIndex].text = newTaskText;
+        localStorage.setItem('todoItems', JSON.stringify(todoItems));
+        display();
+      }
+    });
   }
 };
 
-window.onload = display();
\ No newline at end of file
+window.onload = display();
+
+// Set up form submit event listener to add a new task
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+  const inputText = document.querySelector('.inputText');
+  const text = inputText.value.trim();
+  if (text !== '') {
+    addTask(text);
+    inputText.value = '';
+    ul.innerHTML = ''; // clear the list before re-rendering it
```

Comment:
- [ ] I think that you can safely remove this clear list from this listener, as you already have it at the start of your `display()` function! 🤓 🔧

Reply:
In order to implement the edit functionality with out prompt I have used input element which becomes editable on user clicking on edit option screen shot attached for your kind reference 
![image](https://user-images.githubusercontent.com/117091923/222684719-2f73ee32-c062-4a92-af41-3cb2f6c498d1.png)

Reply:
in my edit event listener I have replaced the prompt with making input field editable when user clicks on edit option user can edit the task. Screen shot attached for your kind reference 
![image](https://user-images.githubusercontent.com/117091923/222685275-3c0457a7-32d1-4535-b234-3c755597d663.png)

Reply:
updated